temporalio/ext/src/util.rs

```diff
@@ -102,3 +102,28 @@ where
         *Box::from_raw(result as _)
     }
 }
+
+pub(crate) struct AsyncCallback {
+    queue: BoxValue<Value>,
+}
+
+// We trust our usage of this across threads. We would use Opaque but we can't
+// box that properly/safely.
+unsafe impl Send for AsyncCallback {}
+unsafe impl Sync for AsyncCallback {}
+
+impl AsyncCallback {
+    pub(crate) fn from_queue(queue: Value) -> Self {
+        Self {
+            queue: BoxValue::new(queue),
+        }
+    }
+
+    pub(crate) fn push<V>(&self, value: V) -> Result<(), Error>
```

Comment:
Is `AsyncCallback` expected to be a one time message only, or is it genuinely intended that the rust side may push values again and again? I mean, would it make sense to make that method capture `self` rather than `&self`?

Reply:
It supports again and again, which is how polling works.